### PR TITLE
[codex] Harden schema, timeline analysis, and workflow naming semantics

### PR DESCRIPTION
**Pull request by thc1006**
## Summary
- harden mission schema timestamps with timezone-aware parsing (`AwareDatetime`)
- harden policy rule for `mission_id` by rejecting whitespace-only values (`trim_space`)
- centralize timeline conflict analysis in compiler and reuse it from MCP server
- make workflow naming collision-resistant under truncation and avoid Argo DAG task-name collisions
- update golden fixtures and architecture/risk docs to match new semantics

## Why
Several semantic checks were duplicated or weakly enforced (timestamp typing, `mission_id` whitespace, timeline analysis drift, truncated-name collisions), which could cause inconsistent behavior between compiler, MCP tools, and rendered artifacts.

## Validation
- `make verify`
- `make test`
- `make eval`

## Scope
This PR intentionally excludes live-cluster validation/RBAC/CI portability updates; those are in a separate follow-up PR.

### CODE REVIEW

File: src/orbital_mission_compiler/compiler.py [hunk 1]
```diff
@@ -21,6 +23,30 @@ def sanitize_k8s_name(name: str, max_len: int = 63) -> str:
...
 
+def _collision_resistant_k8s_name(name: str, max_len: int = 63, hash_len: int = 8) -> str:
+    """Sanitize and preserve uniqueness when truncation is required."""
+    if max_len < hash_len + 2:
+        raise ValueError(f"max_len={max_len} too small for hash_len={hash_len}")
+    normalized = sanitize_k8s_name(name, max_len=253)
+    if len(normalized) <= max_len:
+        return normalized
+    digest = hashlib.sha1(name.encode("utf-8")).hexdigest()[:hash_len]
+    head = normalized[: max_len - hash_len - 1].rstrip("-")
```
[the remark this answers is not in the document]
Comment (thc1006) [reply src/orbital_mission_compiler/compiler.py:34]: Addressed in commit ed8433c. I replaced `hashlib.sha1` with `hashlib.sha256` in `_collision_resistant_k8s_name` to avoid FIPS/runtime tooling issues while keeping deterministic naming.

File: src/orbital_mission_compiler/compiler.py [hunk 2]
```diff
@@ -212,7 +245,7 @@ def render_argo_workflow(intent: WorkflowIntent) -> dict[str, Any]:
             template["affinity"] = affinity
         templates.append(template)
 
-        safe_name = sanitize_k8s_name(step.name)
+        safe_name = sanitize_k8s_name(template_name)
         dag_task: dict[str, Any] = {"name": safe_name, "template": template_name}
         dag_tasks.append(dag_task)
```
[the remark this answers is not in the document]
Comment (thc1006) [reply src/orbital_mission_compiler/compiler.py]: Addressed in commit ed8433c. `template_name` now uses `_collision_resistant_k8s_name(f"step-{idx}-{step.name}")`, and the DAG task name reuses that same bounded/unique value so we do not exceed common name limits or diverge between template/task identifiers.

File: configs/policies/mission_plan.rego [hunk 3]
```diff
@@ -5,7 +5,7 @@ import rego.v1
 default allow := false
 
 deny contains msg if {
-  input.mission_id == ""
+  trim_space(input.mission_id) == ""
   msg := "mission_id must not be empty"
 }
```
[the remark this answers is not in the document]
Comment (thc1006) [reply configs/policies/mission_plan.rego:17]: Addressed in commit ed8433c with explicit policy denies for both missing and `null` `mission_id`, in addition to the whitespace-only check. I also added policy tests for missing/null mission IDs.

File: tests/test_schema_negative.py [hunk 4]
```diff
@@ -39,6 +39,20 @@ def test_reject_invalid_event_type():
...
+
+def test_reject_invalid_timestamp_format():
+    """MissionEvent timestamp must be a valid RFC3339/ISO datetime with timezone."""
+    with pytest.raises(ValidationError):
+        MissionEvent(
+            timestamp="not-a-datetime",
+            event_type=MissionEventType.ACQUISITION,
+            instrument="INST_1",
+            services=[_service()],
+        )
```
[the remark this answers is not in the document]
Comment (thc1006) [reply tests/test_schema_negative.py:53]: Addressed in commit 528b146. I added `test_reject_timezone_naive_timestamp` in `tests/test_schema_negative.py` to explicitly lock in rejection of timezone-naive timestamps (e.g., `2029-01-01T00:00:00`) under `AwareDatetime`.

File: docs/04_architecture.md [hunk 5]
```diff
@@ -44,9 +45,11 @@ Schema validation (Pydantic) and policy validation (OPA/Rego) intentionally over
 
 | Constraint | Schema (schemas.py) | Policy (mission_plan.rego) |
 |---|---|---|
+| timestamp must be parseable + timezone-aware | `MissionEvent.timestamp: AwareDatetime` | — |
 | ACQ must have instrument | `model_validator` raises ValueError | — |
 | DOWNLOAD must not have services | `model_validator` raises ValueError | Rule 7 denies |
 | DOWNLOAD must have visibility | `model_validator` raises ValueError | Rule 8 denies |
+| mission_id must not be missing/null/blank | `field_validator` trims and rejects empty | Rule 1 denies missing/null and trims with `trim_space` |
 | Priority must not be 0 | `Field(ge=0)` allows it | Rule 5 denies it |
```
[the remark this answers is not in the document]
Comment (thc1006) [reply docs/04_architecture.md:53]: Addressed in commit 528b146. I updated `docs/04_architecture.md` to match implementation behavior: the schema validator performs a blank check via `strip()` but does not normalize/trim the stored `mission_id` value.
